File: app/src/main/java/com/blog/demo/image/FrameAnimationActivity.java
package com.blog.demo.image;

import android.app.Activity;
import android.graphics.drawable.AnimationDrawable;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;
import android.widget.CheckBox;
import android.widget.ImageView;

import com.blog.demo.R;

public class FrameAnimationActivity extends Activity implements View.OnClickListener {

    private AnimationDrawable mDrawable;
    private CheckBox cbOneShot;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_image_animation_frame);

        findViewById(R.id.btn_start).setOnClickListener(this);
        findViewById(R.id.btn_stop).setOnClickListener(this);

        cbOneShot = findViewById(R.id.checkbox);

        ImageView imageView = findViewById(R.id.iv_frame);
        mDrawable = (AnimationDrawable) imageView.getDrawable();
    }

    @Override
    public void onClick(View v) {
        if (v.getId() == R.id.btn_start) {
            mDrawable.setOneShot(cbOneShot.isChecked());
            mDrawable.start();
        } else if (v.getId() == R.id.btn_stop) {
            mDrawable.stop();
        }
    }
}

File: app/src/main/java/com/blog/demo/image/animation/FrameAnimationActivity.kt
package com.blog.demo.image.animation

import android.app.Activity
import android.graphics.drawable.AnimationDrawable
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.CheckBox
import android.widget.ImageView
import com.blog.demo.R

class FrameAnimationActivity : Activity(), View.OnClickListener {

    private lateinit var mDrawable: AnimationDrawable
    private lateinit var cbOneShot: CheckBox

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_image_animation_frame)

        findViewById<Button>(R.id.btn_start).setOnClickListener(this)
        findViewById<Button>(R.id.btn_stop).setOnClickListener(this)

        cbOneShot = findViewById(R.id.checkbox)

        val imageView: ImageView = findViewById(R.id.iv_frame)
        mDrawable = imageView.drawable as AnimationDrawable
    }

    override fun onClick(v: View) {
        if (v.id == R.id.btn_start) {
            mDrawable.isOneShot = cbOneShot.isChecked
            mDrawable.start()
        } else if (v.id == R.id.btn_stop) {
            mDrawable.stop()
        }
    }

}